### PR TITLE
[RLlib] `AutoregressiveActionsRLM` overhaul to fix flaky test and simplify.

### LINKED ISSUE #47876: CI test linux://rllib:examples/rl_modules/autoregressive_actions_rl_module is flaky
CI test **linux://rllib:examples/rl_modules/autoregressive_actions_rl_module** is consistently_failing. Recent failures: 
	- https://buildkite.com/ray-project/postmerge/builds/6445#01924ae0-ebd8-436e-85c2-53a980899914
	- https://buildkite.com/ray-project/postmerge/builds/6445#01924a92-23ec-4a0b-aa93-39cfc6af11af
	- https://buildkite.com/ray-project/postmerge/builds/6444#019249b2-2250-4c2c-9298-b2d89e3977f7
	- https://buildkite.com/ray-project/postmerge/builds/6441#01924679-9e62-4843-95ba-f2e360f141f0
	- https://buildkite.com/ray-project/postmerge/builds/6436#019244d5-9e75-41b3-b514-d4c500d647db

DataCaseName-linux://rllib:examples/rl_modules/autoregressive_actions_rl_module-END
Managed by OSS Test Policy

### PR DESCRIPTION
**Pull request by simonsays1980**
## Why are these changes needed?

The autoregressive-actions example was flaky (see #47876) and could be simplified (as PPO only backpropagates through the log-probabilities. This PR suggests a simplified solution that converges very fast (11 iters).

## Related issue number

Closes #47876 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/examples/rl_modules/classes/autoregressive_actions_rlm.py [hunk 1]
```diff
@@ -49,6 +49,7 @@ def setup(self):
 
         # Build the encoder.
         self.encoder = self.catalog.build_encoder(framework=self.framework)
+        self.encoder = self.catalog.build_encoder(framework=self.framework)
```
Comment (sven1977) [inline rllib/examples/rl_modules/classes/autoregressive_actions_rlm.py:52]: why duplicate this line?

File: rllib/examples/rl_modules/classes/autoregressive_actions_rlm.py [hunk 2]
```diff
@@ -126,28 +128,19 @@ def pi(
...
-            # Otherwise, we need to be able to backpropagate through the prior action
-            # that's why we sample from the distribution using the `rsample` method.
-            # TODO (simon, sven): Check, if we need to return the one-hot sampled action
-            # instead of the real-valued one.
-            prior_action = torch.argmax(
-                prior_action_dist._flat_child_distributions[0].rsample(),
-                dim=-1,
-            )
+
+        # Sample the action and reshape.
```
Comment (sven1977) [inline rllib/examples/rl_modules/classes/autoregressive_actions_rlm.py:135]: Nice!

**Review by sven1977 — APPROVED** [review]
LGTM now. Just one nit about the duplicate line.

File: rllib/examples/rl_modules/classes/autoregressive_actions_rlm.py (hunk 1 above)
Comment (simonsays1980) [reply rllib/examples/rl_modules/classes/autoregressive_actions_rlm.py:52]: This is probably from the conflict in a merge.

**Pull request merged**